packages/dd-trace/src/latency_stats.js

```diff
@@ -0,0 +1,156 @@
+const os = require('os')
+const pkg = require('../../../package.json')
+const { decodePathwayContext } = require('../../datadog-plugin-kafkajs/src/hash')
+
+const { LogCollapsingLowestDenseDDSketch } = require('@datadog/sketches-js')
+
+const { LatencyStatsExporter } = require('./exporters/latency-stats')
+
+class AggStats {
+  constructor (aggKey) {
+    this.hash = aggKey.hash
+    this.parentHash = aggKey.parentHash
+    this.edgeTags = aggKey.edgeTags
+    this.edgeLatency = new LogCollapsingLowestDenseDDSketch(0.00775)
+    this.pathwayLatency = new LogCollapsingLowestDenseDDSketch(0.00775)
+  }
+
+  record (checkpoint) {
+    const edgeLatency = checkpoint.metrics.edge_latency
+    const pathwayLatency = checkpoint.metrics.pathway_latency
+    this.edgeLatency.accept(edgeLatency)
+    this.pathwayLatency.accept(pathwayLatency)
+  }
+
+  toJSON () {
+    return {
+      Hash: this.hash,
+      ParentHash: this.parentHash,
+      EdgeTags: this.edgeTags,
+      EdgeLatency: this.edgeLatency.toProto(), // TODO: custom proto encoding
```

Comment:
Are these still TODOs?

Reply:
yeah this was also taken from span_stats, the TODOs are there too